libpromises/evalfunction.c

```diff
@@ -6021,7 +6021,11 @@ static FnCallResult FnCallReadData(ARG_UNUSED EvalContext *ctx,
     }
     else
     {
-        size_max = args->next ? IntFromString(RlistScalarValue(args->next)) : 0;
+        if (RlistLen(args) == 0)
+        {
+            FatalError(ctx, "Function '%s' requires at least one argument", fp->name);
```

Comment:
Maybe ``FatalError`` is a bit brutal here, it will terminate the agent. Since this will only be triggered at runtime, after the policy has been parsed, you may verify the policy on the hub, thinking that everything is fine, but then the agent terminates on your clients because of different conditions. Perhaps returning ``FnFailure()`` is better. What do you think?